### ISSUE TITLE
Popeye Error `Got empty response for: external.metrics.k8s.io/v1beta1`

### ISSUE THREAD

**Issue by nickv2002** [derailed/popeye#249]
**Describe the bug**
Running `popeye` with no args errors out with message:

```
Boom! 💥 unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: Got empty response for: external.metrics.k8s.io/v1beta1
```

I can still run normal commands with `kubectl` and `k9s`. EG:
```
$ kubectl get pods |tail -n1
web-api-6994b5f88-qdxf6  1/1     Running   0   7h47m
```

Unlike the answers suggested for https://github.com/derailed/popeye/issues/171 the offending apiservice is available and working (as far as I know):
```
$ kubectl get apiservices v1beta1.external.metrics.k8s.io
NAME                              SERVICE                                           AVAILABLE   AGE
v1beta1.external.metrics.k8s.io   datadog/datadog-agent-cluster-agent-metrics-api   True        413d
```

**To Reproduce**
Run `popeye` with no args, see error.

**Expected behavior**
Popeye should work

**Versions (please complete the following information):**
 - OS: Ubuntu Linux
 - Popeye 0.11.1

```
$ kubectl version --short=true
Client Version: v1.23.7
Server Version: v1.24.8-eks-ffeb93d
```

**Additional context**
Is there any CLI arg I could use to tell Popeye to ignore this `apiservice`?

**Comment by Jeansen** [comment]
Got something similar. In my cast it is:

`Got empty response for: custom.metrics.k8s.io/v1beta1`

Kubernetes is a bit older, 1.20.x (on premise, Rancher 2) and 1.21.x (AWS)

**Comment by Jeansen** [comment]
When I delete the "offending" apiservice, popeye complains it is not there. Otherwise, it is also available.

**Comment by otherguy** [comment]
Any news here? https://github.com/derailed/popeye/issues/258 seems related but there is no response there either 🤔

**Comment by ltartarini** [comment]
Same issue for me. I am running EKS 1.25 with `prometheus-adapter`:
```
kubectl get apiservice v1beta1.external.metrics.k8s.io                                     
NAME                              SERVICE                         AVAILABLE   AGE
v1beta1.external.metrics.k8s.io   monitoring/prometheus-adapter
```
![Screenshot 2023-06-29 at 12 16 52](https://github.com/derailed/popeye/assets/44971767/74f960d8-6a94-4cd1-8356-de2fbea1f0ef)

**Comment by derailed** [comment]
Thank you all for piping in! 
I think this might be related to [this issue](https://github.com/kubernetes/kubernetes/pull/115978).  
I'll bump the dependencies on the next drop and see if we're happier...

**Comment by otherguy** [comment]
Thanks @derailed! Is there any progress on this?

**Comment by rodrigc** [comment]
@nickv2002 do you think that this [commit to Keda](https://github.com/kedacore/keda/commit/8c9dd7fe463338a9794bba7f498d847c297eb419) ,
mentioned here: https://github.com/kedacore/keda/issues/4224#issuecomment-1625077067
will address this problem?

**Comment by JorTurFer** [comment]
This problem looks related with this issue in the tooling: https://github.com/kubernetes-sigs/custom-metrics-apiserver/issues/146

**Comment by nickv2002** [comment]
I'm unable to confirm as I've left the company where I experienced this issue. Thanksfor all the work fixing it though!

**Comment by otherguy** [comment]
I don't think this is fixed:

```
$ kubectl version --short
{"kind":"APIResourceList","apiVersion":"v1","groupVersion":"external.metrics.k8s.io/v1beta1","resources":[]}
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.27.2
Kustomize Version: v5.0.1
Server Version: v1.26.5-gke.1200
```

```
popeye --namespace default
E0816 13:19:45.270407   37039 memcache.go:255] couldn't get resource list for external.metrics.k8s.io/v1beta1: Got empty response for: external.metrics.k8s.io/v1beta1
 ___     ___ _____   _____                       D          .-'-.
| _ \___| _ \ __\ \ / / __|                       O     __| K    `\
|  _/ _ \  _/ _| \ V /| _|                         H   `-,-`--._   `\
|_| \___/_| |___| |_| |___|                       []  .->'  X     `|-'
  Biffs`em and Buffs`em!                            `=/ (__/_       /
                                                      \_,    `    _)
                                                         `----;  |


Boom! 💥 unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: Got empty response for: external.metrics.k8s.io/v1beta1
```

**Comment by JorTurFer** [comment]
Hi,
I didn't say that it's solved everywhere, I pointed the upstream (kubernetes client) issue and I said that the fix is already merged in the upstream. How fast is the fix in the upstream propagated depends on each tool, if this project uses that dependency, this project has to bump it to fix the issue.

**Disclaimer**: I have no idea about this repo, I'm a maintainer of KEDA and I found this issue because it was linked to an issue in KEDA. If you are using KEDA and you are affected by this, you can bump KEDA to latest version, it will mitigate the issue even though the tools are still using affected versions

**Comment by otherguy** [comment]
@JorTurFer got it, thanks! That is indeed very helpful! I naively assumed popeye was just invoking the installed `kubectl` but this makes more sense.

Unfortunately this issue isn't caused by KEDA for me, it's the standard GKE metrics adapter.

@derailed is it possible to bump this dependency?

**Comment by JorTurFer** [comment]
This is the commit in client-go: https://github.com/kubernetes/client-go/commit/0bc91705fa50a531b8a1ee9dca3ef063ca71bb0c

**Comment by JorTurFer** [comment]
Any project written in Golang who access to K8s api is can be affected because the way of going programmatically, it's using that dependency instead of kubectl.
Any operator can be affected for example

**Comment by rbjorklin** [comment]
The referenced fix is only for the memory cached client. The fix is unfortunately not as simple as just bumping the dependency.

**Comment by compiledtofu** [comment]
Hi! Is there a current workaround or solution for this issue? My team is trying to use this tool but we're all stuck with this error. We have installed the latest version of popeye

**Comment by rbjorklin** [comment]
I didn't try it but it should be possible to create "dummy" objects of the type that is throwing errors so that it doesn't return an empty list. If you give it a try please report back :)

**Comment by derailed** [comment]
Fixed v0.20.0
